_gsocblogs/2024/blog_Key4hep_BraulioRivas.md

```diff
@@ -0,0 +1,131 @@
+
+However, eede (which was previously called dmx) had a problem. It only allowed to visualize a specific collection/datatype (defined above as a step) named `MCParticle`. This is a problem because the data model has many other collections that are relevant when analyzing an event, and being limited to only one type may lead to ignore important anomalies in an event that would be otherwise hard to detect using other techniques. This is why the project "Any collection in Data Model Explorer" was created. The goal was to incorporate new features for eede, adding a whole new set of collections that eede can visualize (according to the EDM4hep event data model). This way, we can visualize not only these called collections, but also explore many types of relations that exists between them. Here we have a diagram that better represents the types of collections and existing relations.
+
+![EDM4hep Model](https://raw.githubusercontent.com/key4hep/EDM4hep/main/doc/edm4hep_diagram.svg)
+
+# Proposed solution
+
+The original [goal](https://hepsoftwarefoundation.org/gsoc/2024/proposal_Key4hepAnyCollectionInDataModelExplorer.html) was to allow to visualize all the remaining **datatypes** in the EDM4hep, allow to filter collections and test on simulated FCC (Future Circular Collider) events. However, due to time constraints, together with mentors, we decided to expand the functionality of eede to a smaller set of collections than initially thought. After some discussion, we defined these goals:
+
+1. Implement loading of Cluster, ParticleID, Reconstructed Particle, Vertex, and Track collections types by allowing users to upload JSON file(s).
+2. Design appropriate graphical representations for every collection type defined by analyzing each type's properties.
+3. Develop tools to filter, manipulate, highlight, and group related collections and objects to allow users to explore the visualized events easily.
+4. Conduct testing to cover the new features using data from FCC events, either pre-generated or newly generated, to ensure that every version of EDM4hep can be handled correctly.
+5. Incorporate CI/CD rules to automate the deployment of dmX, enabling users to always access the tool's latest version.
+
+# Coding period
+
+**CI/CD rules**
+
+At the very beginning, I started with new CI rules to automate development for eede. We wanted to do a few things: deploy a
+preview of any pull request, so when making changes, in some way an URL will be available to use eede with the changes from the
+PR, so its easier to review changes. It was harder than expected, but we finally were able to configure Github Actions to make
+it work. Also, it was very simple to later add tests whenever running these previews, or before merging to main.
```

Comment:
I think this would be a good opportunity to actually mention the challenges we ran into and how we solved them in the end.